packages/shared/src/validation/addressValidation.ts

```diff
@@ -108,25 +108,32 @@ export const validateBitcoinRegtestAddress: AddressValidator = (
 export const validateChainAddress = (
   chain: Chain,
   address: string,
-  isMainnet = true,
+  network: ChainflipNetwork,
 ): boolean => {
   switch (chain) {
     case 'Bitcoin':
-      return isMainnet
-        ? validateBitcoinMainnetAddress(address)
-        : validateBitcoinTestnetAddress(address) ||
-            validateBitcoinRegtestAddress(address);
+      switch (network) {
```

Comment:
looks like an opportunity for another map: `Record<ChainflipNetwork, Record<ChainflipChain, (addr: string) => boolean>>`

Reply:
it looks like that indeed, but im too lazey to do right now 😇